### DISCUSSION TITLE
[newbie] Can I add "comments" section on my website created with Beaker?

### DISCUSSION THREAD

**Discussion by akavel** [beakerbrowser/beaker#1688] [General]
I would like to be able to receive comments from all people, not only those I follow - is that possible? Though ideally also be able to moderate the comments...

**Comment by pfrazee** [comment]
I've seen people do it using [PeerSockets](https://docs.beakerbrowser.com/apis/beaker.peersockets). You would have to leave your tab open to listen for messages from visitors. We're discussing a "PeerServers" stack which would allow you to listen for comments via a program that runs in the background, so then you don't need to keep a tab open.

Another thing we're looking at is creating a distributed indexing solution so that people who write comments socially can have their comments easily found. I think that approach will scale much better. We'll keep everyone updated on it.

**Comment by brechtcs** [comment]
I created a comment system called [Feedback](hyper://9bda2b8b224a3a4fef10d6302d07de36e118e5065ea9bc16359091ba968bf13b/) which includes a gossip function using PeerSockets. While you can't see *all* the comments made on a certain page, it does retrieve comments by the contacts of all people currently viewing the page (making one extra hop in the network basically). 

Disclaimer: I wrote all the code in the last 24 hours, so there's problably still bugs and rough edges to the implementation.

**Reply by akavel** [reply]
❤️  ...just, uh oh, do you maybe have a link to it that you'd be willing to share? :) I'm ok with it being hacky, I feel the "hypernet" has this cool geocities-like vibe to it now, and I find that the builtin editor makes JS experimentation feel super easy and really encouraged to me :)

**Reply by brechtcs** [reply]
Interesting, it seems Github markdown refuses to render links using the `hyper://` protocol. So, here it is in plain text:

```
hyper://9bda2b8b224a3a4fef10d6302d07de36e118e5065ea9bc16359091ba968bf13b/
```
